core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java

```diff
@@ -1877,6 +1885,19 @@ private SqlDialect nonOrdinalDialect() {
         + "FROM \"foodmart\".\"employee\"");
   }
 
+  /** Test case for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-5724">[CALCITE-5724]
+   * Test the ORDER BY is properly removed when all literal nodes are filtered out
+   * in a dialect that doesn't support ORDER BY literals</a>.
+   */
+  @Test void testRemoveEmptyOrderByClause() {
+    String query = "select 5, \"product_id\" from \"product\"\n"
+        + "order by 1";
+    final String expected = "SELECT 5, product_id\n"
+        + "FROM foodmart.product";
+    sql(query).withBigQuery().ok(expected);
```

Comment:
i'd chain `.withMySql().ok(expectedMySql)` just so we see what happens in the 'normal' case.